test/lit.cfg

```diff
@@ -0,0 +1,480 @@
+    tool_name = tool_match.group(4)
+    tool_path = lit.util.which(tool_name, tool_dirs)
+    if not tool_path:
+        # Warn, but still provide a substitution.
+        lit_config.note('Did not find ' + tool_name + ' in ' + tool_dirs)
+        tool_path = flang_tools_dir + '/' + tool_name
+    config.substitutions.append((pattern, tool_pipe + tool_path))
+
+# Passing in custom flags.
+flags = lit_config.params.get('flags', '')
+flags = flags.replace(',', ' ')
+config.substitutions.append(('%flags', '\"' + flags + '\"'))
+
+# Option to keep files generated while running tests.
+keep_file = lit_config.params.get('keep', None)
+if keep_file != None:
+  config.substitutions.append(('%keep', '1'))
+else:
+  config.substitutions.append(('%keep', ''))
+
+
+###
+
+# Set available features we allow tests to conditionalize on.
+#
+# Enabled/disabled features
+#if config.flang_staticanalyzer != 0:
+#    config.available_features.add("staticanalyzer")
+
+# As of 2011.08, crash-recovery tests still do not pass on FreeBSD.
+if platform.system() not in ['FreeBSD']:
+    config.available_features.add('crash-recovery')
+
+# Shell execution
+if execute_external:
+    config.available_features.add('shell')
+
+# For tests that require Darwin to run.
+# This is used by debuginfo-tests/*block*.m and debuginfo-tests/foreach.m.
+if platform.system() in ['Darwin']:
+    config.available_features.add('system-darwin')
+elif platform.system() in ['Windows']:
+    # For tests that require Windows to run.
+    config.available_features.add('system-windows')
+
+# ANSI escape sequences in non-dumb terminal
+if platform.system() not in ['Windows']:
+    config.available_features.add('ansi-escape-sequences')
+
+# Capability to print utf8 to the terminal.
+# Windows expects codepage, unless Wide API.
+if platform.system() not in ['Windows']:
+    config.available_features.add('utf8-capable-terminal')
+
+# Native compilation: Check if triples match.
+# FIXME: Consider cases that target can be executed
+# even if host_triple were different from target_triple.
+if config.host_triple == config.target_triple:
+    config.available_features.add("native")
+
+# Case-insensitive file system
+def is_filesystem_case_insensitive():
+    handle, path = tempfile.mkstemp(prefix='case-test', dir=config.test_exec_root)
+    isInsensitive = os.path.exists(
+        os.path.join(
+            os.path.dirname(path),
+            os.path.basename(path).upper()
+            ))
+    os.close(handle)
+    os.remove(path)
+    return isInsensitive
+
+if is_filesystem_case_insensitive():
+    config.available_features.add('case-insensitive-filesystem')
+
+# Tests that require the /dev/fd filesystem.
+if os.path.exists("/dev/fd/0") and sys.platform not in ['cygwin']:
+    config.available_features.add('dev-fd-fs')
+
+# Not set on native MS environment.
+if not re.match(r'.*-win32$', config.target_triple):
+    config.available_features.add('non-ms-sdk')
+
+# Not set on native PS4 environment.
```

Comment:
Similar comment above about only including needed features, etc.

Reply:
This file has being removed

Reply:
https://github.com/flang-compiler/f18/pull/861